src/lib/files.c

```diff
@@ -169,6 +169,32 @@ static bool get_parent_disk_devno(char *path, dev_t *diskdevno)
         return true;
 }
 
+bool cbm_is_file_empty(char *path)
+{
+        int fd = -1;
+        struct stat st = { 0 };
+        ssize_t length = -1;
+
+        if (!nc_file_exists(path)) {
+                return false;
+        }
+
+        fd = open(path, O_RDONLY);
+        if (fd < 0) {
+                return false;
+        }
+
+        if (fstat(fd, &st) != 0) {
+                close(fd);
+                return false;
+        }
+
+        length = st.st_size;
+        close(fd);
+
+        return length == 0;
+}
+
```

Comment:
So I think the function is probably better as cbm_file_has_content.

The reason I think this is that the only use so far is a negative check and that a potential user might expect the current function to return false only when the file exists and has content. If the return is only true when the length != 0 then this would resolve the latter potential confusion.

Reply:
done. thanks.